### PR TITLE
Provide randomness in Move UTs

### PR DESCRIPTION
**Pull request by benr-ml**
## Description 

Provide randomness in Move UTs: instantiate `RandomGenerator` from a given seed or a random one.

## Test plan 

How did you test the new or updated feature?

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:

### CODE REVIEW

**Review by tnowacki — COMMENTED** [review]
LGTM but since @tzakian and I have had such a fun morning with unit tests (context: some bugs with test_scenario), going to get his eyes on it too

**Review by tzakian — APPROVED** [review]
Overall looks good to me as well! Don't think this should run-up against the issue we found either :)

File: crates/sui-framework/packages/sui-framework/sources/random.move [hunk 1]
```diff
@@ -306,4 +306,21 @@ module sui::random {
...
+    }
+
+    #[test_only]
+    /// Random generator from a given seed.
+    public fun new_generator_from_seed(seed: vector<u8>): RandomGenerator {
+        RandomGenerator { seed, counter: 0, buffer: vector[] }
+    }
+
+    #[test_only]
+    native fun generate_rand_seed(): vector<u8>;
```
Comment (sblackshear) [inline crates/sui-framework/packages/sui-framework/sources/random.move:325]: Could we give this a scarier name (include `for_testing` or similar) to make sure no one accidentally removes the `test_only`?

Comment (benr-ml) [reply crates/sui-framework/packages/sui-framework/sources/random.move:325]: renamed in https://github.com/MystenLabs/sui/pull/17225

**Pull request merged**